frontend/app/src/components/nft/NftGalleryItem.vue

```diff
@@ -11,43 +13,133 @@ const props = defineProps({
 });
 
 const { item } = toRefs(props);
+const { addToListImagesDomain, shouldRenderImage } = useNftsStore();
+
+const frontendStore = useFrontendSettingsStore();
+
+const { whitelistedDomainsForNftImages } = storeToRefs(frontendStore);
+const { updateSetting } = frontendStore;
+
+const imageUrlSource: ComputedRef<string | null> = computed(
+  () => get(item).imageUrl
+);
+
+watch(
+  imageUrlSource,
+  image => {
+    if (image) {
+      addToListImagesDomain(image);
```

Comment:
Why do we need this?

Reply:
Can the user add a domain that they know, which doesn't already appear in the gallery?

For this in general I would turn tracking and auto-suggestion off and allow the users to input anything that checks out as a valid domain regardless.

For specifics from the screenshot, I can see that you already added a click to trust so that would be enough. 

Also, maybe for the whitelist mode, we should add a confirmation step, just to avoid accidentally whitelisting domains, since this might have implications

Reply:
Yes they can add the domain, outside from the given list.
For the confirmation, yeah I think it makes sense.